analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -48,11 +60,11 @@ public override ProgramState ConditionEvaluated(SymbolicContext context)
         {
             if (context.State[operation].Constraint<BoolConstraint>().Kind == ConstraintKind.True)
             {
-                trueOperations.Add(operation);
+                trueOperations[operation] = context.Block;
             }
             else
             {
-                falseOperations.Add(operation);
+                falseOperations[operation] = context.Block;
             }
         }
         return base.ConditionEvaluated(context);
```

Comment:
Same here, to cut the need for dependency exploration. We don't want any functionality of the base
```suggestion
        return context.State;
```